packages/strapi-utils/lib/content-types.js

```diff
@@ -67,7 +67,12 @@ const getWritableAttributes = (model = {}) => {
 };
 
 const getNonVisibleAttributes = model => {
-  return _.uniq([model.primaryKey, ...getTimestamps(model), ...NON_VISIBLE_ATTRIBUTES]);
+  return _.uniq([
+    model.primaryKey,
+    ...NON_VISIBLE_ATTRIBUTES,
+    ...getTimestamps(model),
+    ...getNonWritableAttributes(model),
```

Comment:
So if an attribute is non-writable, then it shouldn't be visible too, right?

Reply:
Yep, more details answered lower